CT-4145 free text other option

willmcb opened the pull request:
## Description
Adds a text field to add more detail when a user chooses 'other' for outcome reasons.

Text field should appear when other option is selected.

## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed


### Screenshots
Before:
<img width="642" alt="Screenshot 2022-03-28 at 16 23 01" src="https://user-images.githubusercontent.com/13377553/160431863-f99a2905-db76-4599-8573-d0558e78db10.png">


After:
<img width="636" alt="Screenshot 2022-03-28 at 16 22 31" src="https://user-images.githubusercontent.com/13377553/160431791-4e58c0ef-5433-4221-9d94-bdbc14fdf968.png">

Also on show page:
<img width="1063" alt="Screenshot 2022-03-28 at 16 23 37" src="https://user-images.githubusercontent.com/13377553/160432010-ad828ad0-e0d1-4cb0-87c7-353f7d309fa1.png">


### Related JIRA tickets
[CT-4245](https://dsdmoj.atlassian.net/browse/CT-4145)


### Manual testing instructions
- take a case to closure.
- when selecting reasons for the closure. Select other.
- the "Please provide more details" box should appear
- enter your a reason for selecting other
- submit the form.

Note:
- new text box should not appear if other not selected
- field should not submit if other selected and no further reason given.

Commit:
Add some js and form fields for other overturned

Commit:
Structure sql update

Commit:
Update JS

Commit:
Add validations

Commit:
Add model specs for other_overturned validations

Commit:
Add correct label to other overturned field

Commit:
Add other overturned to feature spec + show page

digitalali-moj reviewed (commented):
Looks good on f/e, just couple of questions with no changes needed.

Commit:
Add check for sar_ir to stop method missing warning

Commit:
Add better way to check if other reason present

Commit:
Fix spec

Commit:
Fix specs

Commit:
Fix spec

Commit:
Us checkbox id to make JS more robust

Commit:
Make name of new field more meaningful

was "other_overturned" which didn't make much sense.

Is now "other_option_details"

Commit:
Fix spec

digitalali-moj reviewed (commented):
Questions really, with one minor comment

digitalali-moj reviewed (approved):
Nice work! Looks good on f/e
